rfcs/2021-06-02-7709-helm-deprecate-custom-config-keys.md

```diff
@@ -0,0 +1,97 @@
+# RFC 7709 - 2021-06-02 - Helm: Deprecate "custom" configuration keys
+
+Today the templates that user's leverage to generate their Vector config are separate from the ones used to generate the "default" Vector configuration.
+While these are not very complex in their functionality they do duplicate much of the same logic.
+
+## Scope
+
+This RFC will cover only the existing "default" `sinks` and `sources` defined in our Helm charts today.
+
+vector-agent:
+
+- `kubernetes_logs`
+- `vector`
+- `internal_metrics`
+- `host_metrics`
+- `prometheus_exporter`
+
+vector-aggregator:
+
+- `vector`
+- `internal_metrics`
+- `prometheus_exporter`
+
+## Motivation
+
+We're duplicating the templating used to generate Vector's configuration file in Helm based deployments. While I was working on transitioning the
+currently TOML config to YAML realized the configuration building had several unexpected complications due to the separation between Vector's provided
+"default" configurations (listed in the Scope section) and a user provided configuration located under the `sources`, `transforms`, and `sinks` keys.
+
+## Internal Proposal
+
+The "custom" `values.yaml` keys be deprecated and replaced with default values under the `sources`, `transforms`, and `sinks` keys that users configure
+for their provided Vector configuration. An example is below:
+
+```yaml
+-  internalMetricsSource:
+-    enabled: true
+-    sourceId: internal_metrics
+-    config: {}
+-  hostMetricsSource:
+-    enabled: true
+-    sourceId: host_metrics
+-    config:
+-      filesystem:
+-        devices:
+-          excludes: [binfmt_misc]
+-        filesystems:
+-          excludes: [binfmt_misc]
+-        mountpoints:
+-          excludes: ["*/proc/sys/fs/binfmt_misc"]
++  sources:
++    internal_metrics:
++      type: internal_metrics
++    host_metrics:
++      type: host_metrics
++      filesystem:
++        devices:
++          excludes: ["binfmt_misc"]
++        filesystems:
++          excludes: ["binfmt_misc"]
++        mountpoints:
++          excludes: ["*/proc/sys/fs/binfmt_misc"]
+```
```

Comment:
I think you would want to separate the configuration here into a "this existing pattern goes from <this> to <that>" structure to clarify the change.

Reply:
Not super clear what you mean by this, the the before to after is existing to the "generic configuration keys"